### DISCUSSION TITLE
Ideas about migrating sanni cartreader project to a 32-bit MCU?

### DISCUSSION THREAD

**Discussion by Hedda** [sanni/cartreader#103] [Ideas & Suggestions]
I don't have the coding skills myself but wondering if anyone has thoughts on porting this cartreader project to a 32-bit MCU?

SoCs and development boards with low-cost faster 32-bit MCUs with relatively higher-performance are getting more common.

Perhaps could be possible to base a port on HAL(s) from Marlin Firmware as they already ported from 8-bit AVR to 32-bit HALs?

https://github.com/MarlinFirmware/Marlin/tree/bugfix-2.0.x

Marlin Firmware is a famous open source project well know for being ported from Atmel's 8-bit AVRs to multiple 32-bit MCUs.

https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/README.md

There are many MCU options but FYI, as posted in the other thread, RetroArch/Libretro developers have announced that they are working on a new open source game cartridge hardware adapter which will be based on an 32-bit STM32 MCU from ST -> https://github.com/sanni/cartreader/discussions/101

As I understand, the main benifit with the different STM32 families of MCUs is that they are somewhat backwards compatible.

**Comment by sanni** [comment]
I will re-evaluate STM32 once the RetroArch/Libretro developers have implemented all the different consoles into their project. Maybe they can change my mind about STM32.
